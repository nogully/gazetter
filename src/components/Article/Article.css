section {
  display: flex;
  justify-content: space-between;
  margin: 10px 0px;
}

.article-link {
  color: #0084b4;
  font-family: 'Open Sans', sans-serif;
  text-decoration: none;
  text-align: left;
  font-size: 1rem;
}

.article-link:hover {
  color: #00aced;
}

.article-link:active {
  color: #1dcaff;
  text-decoration: underline;
}

h5 {
  display: inline-block;
  text-align: right;
  color: #0084b4;
  font-family: 'Open Sans', sans-serif;
  font-size: 1rem;
}

img {
  border-bottom: 10px;
}

.bookmark {
  border: 1px solid #c0deed;
  background-color: #FFF;
  box-shadow: 5px 5px 5px #CCC;
}

.bookmark .article-link {
  color: #444;
}

#twitter-icons {
  margin-left: 15px;
}

#bookmark {
  color: #0084b4;
}