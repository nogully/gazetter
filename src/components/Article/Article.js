import React from 'react';

const Article = ({tweet}) => {
  return (
    <article className="tweet" key={tweet.id}>
      { tweet.entities.media ? <img src={tweet.entities.media[0].media_url} alt="tweet"/> : null }
      <h2>{tweet.user.name}</h2> 
      <p>{tweet.full_text}</p> 
      { tweet.entities.urls.length ? <a href={tweet.entities.urls[0].expanded_url} target="_blank">Read</a> : null}
      <h5><i className="fas fa-heart"></i> {tweet.favorite_count}</h5>
    </article>

  )
}

export default Article;