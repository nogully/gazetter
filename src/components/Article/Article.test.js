/* eslint-disable */
import React from 'react';
import { shallow } from 'enzyme';
import Article  from './Article';

describe('Article', () => {
  it('should match its snapshot', () => {
    const mockTweet = { id: 1234,
                full_text: 'Jhun named best teacher ever', 
                entities: { urls: [{ expanded_url: 'http://wow.com'}], 
                            media: [{ media_url: 'http://awesome.com'}] }, 
                user: { screen_name: 'nytimes' }, 
                retweeted_status: 345,
                favorite_count: 456 }
    const wrapper = shallow( <Article tweet={ mockTweet } /> )
    expect(wrapper).toMatchSnapshot();
  })

})