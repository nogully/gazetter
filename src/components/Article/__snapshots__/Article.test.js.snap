// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article should match its snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Article
    tweet={
        Object {
            "entities": Object {
              "media": Array [
                Object {
                  "media_url": "http://awesome.com",
                },
              ],
              "urls": Array [
                Object {
                  "expanded_url": "http://wow.com",
                },
              ],
            },
            "favorite_count": 456,
            "full_text": "Jhun named best teacher ever",
            "id": 1234,
            "retweeted_status": 345,
            "user": Object {
              "screen_name": "nytimes",
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "1234",
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="tweet"
          onClick={[Function]}
          src="http://awesome.com"
/>,
        <h2
          id={undefined}
          onClick={[Function]}
/>,
        <p
          onClick={[Function]}
>
          Jhun named best teacher ever
</p>,
        <section>
          <p
                    className="article-link"
                    id={1234}
                    onClick={undefined}
          >
                    <i
                              className="fas fa-plus-circle"
                    />
                     Bookmark
          </p>
          <h5
                    className="favorite"
          >
                    <i
                              className="fas fa-retweet"
                              id="twitter-icons"
                    />
                     
                    <i
                              className="fas fa-heart"
                              id="twitter-icons"
                    />
                     
                    456
          </h5>
</section>,
      ],
      "className": "tweet undefined",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "tweet",
          "onClick": [Function],
          "src": "http://awesome.com",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "id": undefined,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Jhun named best teacher ever",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Jhun named best teacher ever",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="article-link"
              id={1234}
              onClick={undefined}
>
              <i
                            className="fas fa-plus-circle"
              />
               Bookmark
</p>,
            <h5
              className="favorite"
>
              <i
                            className="fas fa-retweet"
                            id="twitter-icons"
              />
               
              <i
                            className="fas fa-heart"
                            id="twitter-icons"
              />
               
              456
</h5>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <i
                  className="fas fa-plus-circle"
/>,
                " Bookmark",
              ],
              "className": "article-link",
              "id": 1234,
              "onClick": undefined,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fas fa-plus-circle",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              " Bookmark",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <i
                  className="fas fa-retweet"
                  id="twitter-icons"
/>,
                " ",
                undefined,
                <i
                  className="fas fa-heart"
                  id="twitter-icons"
/>,
                " ",
                456,
              ],
              "className": "favorite",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fas fa-retweet",
                  "id": "twitter-icons",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              " ",
              undefined,
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fas fa-heart",
                  "id": "twitter-icons",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              " ",
              456,
            ],
            "type": "h5",
          },
        ],
        "type": "section",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "1234",
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="tweet"
            onClick={[Function]}
            src="http://awesome.com"
/>,
          <h2
            id={undefined}
            onClick={[Function]}
/>,
          <p
            onClick={[Function]}
>
            Jhun named best teacher ever
</p>,
          <section>
            <p
                        className="article-link"
                        id={1234}
                        onClick={undefined}
            >
                        <i
                                    className="fas fa-plus-circle"
                        />
                         Bookmark
            </p>
            <h5
                        className="favorite"
            >
                        <i
                                    className="fas fa-retweet"
                                    id="twitter-icons"
                        />
                         
                        <i
                                    className="fas fa-heart"
                                    id="twitter-icons"
                        />
                         
                        456
            </h5>
</section>,
        ],
        "className": "tweet undefined",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "tweet",
            "onClick": [Function],
            "src": "http://awesome.com",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "id": undefined,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Jhun named best teacher ever",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Jhun named best teacher ever",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="article-link"
                id={1234}
                onClick={undefined}
>
                <i
                                className="fas fa-plus-circle"
                />
                 Bookmark
</p>,
              <h5
                className="favorite"
>
                <i
                                className="fas fa-retweet"
                                id="twitter-icons"
                />
                 
                <i
                                className="fas fa-heart"
                                id="twitter-icons"
                />
                 
                456
</h5>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <i
                    className="fas fa-plus-circle"
/>,
                  " Bookmark",
                ],
                "className": "article-link",
                "id": 1234,
                "onClick": undefined,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fas fa-plus-circle",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                " Bookmark",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <i
                    className="fas fa-retweet"
                    id="twitter-icons"
/>,
                  " ",
                  undefined,
                  <i
                    className="fas fa-heart"
                    id="twitter-icons"
/>,
                  " ",
                  456,
                ],
                "className": "favorite",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fas fa-retweet",
                    "id": "twitter-icons",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                " ",
                undefined,
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fas fa-heart",
                    "id": "twitter-icons",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                " ",
                456,
              ],
              "type": "h5",
            },
          ],
          "type": "section",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
